### PR TITLE
Implement CI as a GitHub Action

### PR DESCRIPTION
**Pull request by locks**
I haven't removed the Circle CI configuration yet because I wanted to confirm with maintainers that this is the way to go.
Closes https://github.com/mtgred/netrunner/issues/6493.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches: ["master"]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: lein deps
```
Comment (NoahTheDuke) [inline .github/workflows/ci.yml:16]: Unnecessary as all calls to `lein` will resolve deps first.

Comment (locks) [reply .github/workflows/ci.yml:16]: Interesting. It was like that in the default Clojure workflow and I think all the ones I saw during my research. Removing this step!

Comment (locks) [reply .github/workflows/ci.yml]: @NoahTheDuke people might do this so as not to get the dependency spam when running other steps:
image

I'd suggest keeping it separate even if not strictly necessary to make it easier to troubleshoot.

Comment (NoahTheDuke) [reply .github/workflows/ci.yml]: It looks like the cache is only being set when the tests don't fail, so maybe include the fix in this PR to see what happens.

**Comment by NoahTheDuke** [comment]
Looks like this works, so go ahead and remove the CircleCI folder. rip in pepperonis circle!

**Pull request merged**